gqcpy/include/interfaces.hpp

```diff
@@ -699,4 +700,145 @@ void bindSimple2DMInterface(Class& py_class) {
+ *  
+ *  @param py_class             The Pybind11 `class_` that should obtain APIs related to `QCModel::R/U/GHF`.
+ */
+template <typename Class>
+void bindQCModelHartreeFockInterface(Class& py_class) {
+
+    // The C++ type corresponding to the Python class.
+    using Type = typename Class::type;
+    using Hamiltonian = typename Type::Hamiltonian;
+    using Scalar = typename Type::Scalar;
+
+    py_class
+        .def(
+            "calculateOrthonormalBasis1DM",
+            [](const Type& parameters) {
+                return parameters.calculateOrthonormalBasis1DM();
+            },
+            "Return the 1-DM expressed in an orthonormal basis related to these optimal HF parameters.")
+
+        .def(
+            "calculateScalarBasis1DM",
+            [](const Type& parameters) {
+                return parameters.calculateScalarBasis1DM();
+            },
+            "Return the HF 1-DM in the scalar/AO basis related to these optimal parameters.")
+
+        .def(
+            "calculateStabilityMatrices",
+            [](const Type& parameters, const Hamiltonian& hamiltonian) {
+                return parameters.calculateStabilityMatrices(hamiltonian);
+            },
+            "Return the HF stability matrices parameters.")
+
+        .def("expansion",
+             &Type::expansion,
+             "Return the coefficient matrix that expresses every spatial orbital (as a column) in its underlying scalar basis.")
+
+        .def(
+            "numberOfElectrons",
+            [](const Type& parameters) {
+                return parameters.numberOfElectrons();
+            },
+            "Return the number of electrons that these HF model parameters describe, i.e. the number of occupied orbitals.")
+
+        .def("occupiedOrbitalEnergies",
+             &Type::occupiedOrbitalEnergies,
+             "Return the orbital energies of the occupied orbitals.")
+
+        .def("orbitalEnergies",
+             &Type::orbitalEnergies,
+             "Return all of the orbital energies.")
+
+        .def("virtualOrbitalEnergies",
+             &Type::virtualOrbitalEnergies,
+             "Return the orbital energies of the virtual orbitals.");
+}
+
+
+/**
+ *  Add Python bindings for some APIs related to `complex R/U/GHF Stability matrices`.
+ * 
+ *  @tparam Class               The type of the Pybind11 `class_` (generated by the compiler).
+ *  
+ *  @param py_class             The Pybind11 `class_` that should obtain APIs related to `R/U/GHFStabilityMatrices`.
+ */
+template <typename Class>
+void bindQCModelHartreeFockComplexStabilityInterface(Class& py_class) {
+
+    // The C++ type corresponding to the Python class.
+    using Type = typename Class::type;
+    using Scalar = typename Type::Scalar;
+
+    py_class
+        .def(
+            "internal",
+            [](const Type& stability_matrices) {
+                return stability_matrices.internal();
+            },
```

Comment:
```suggestion
            &Type::internal,
```

Reply:
This can't be rewritten like that. I get the error that there is no matching function for `def`, for all six classes of `StabilityMatrices`(real/complex R/U/GHF).

![Screenshot 2020-12-07 at 14 09 42](https://user-images.githubusercontent.com/54795591/101354799-011f2400-3896-11eb-8e74-b06661163688.png)

Reply:
I see, that's probably due to the overloading.